add golang runtime lifecycle

freedomkk-qfeng opened the pull request.

freedomkk-qfeng committed:
add golang runtime lifecycle

inntran commented on runtime/golang.md:
```diff
@@ -0,0 +1,22 @@
+# Golang
+
+### 生命周期
+Golang 的生命周期策略为，支持最新的2个大版本。例如，Go 1.5在Go 1.7发布之前一直受支持，Go 1.6在Go 1.8发布之前一直受支持。
```
中英文空格间策略希望保持一致

freedomkk-qfeng replied:
统一为中英文间保持一个空格吧，我改一下

inntran commented on runtime/golang.md:
```diff
@@ -0,0 +1,22 @@
+### 兼容性
+Golang 1.x 版本保持向下兼容，因此任何低版本的 Go 语言程序都可以直接通过新版本直接编译来提升性能和修复 bug。
+
+但如果引入了 cgo 则可能要考虑 gcc 的兼容性，在此常见下应该单独考虑。
```
场景？

freedomkk-qfeng replied:
错别字了，改

inntran commented on runtime/golang.md:
```diff
@@ -0,0 +1,22 @@
+
+但如果引入了 cgo 则可能要考虑 gcc 的兼容性，在此常见下应该单独考虑。
+
+Golang 兼容性指 Go 语言本身，并不包括 Go 相关工具的兼容性。在某些版本中部分 go 相关工具可能存在兼容性破坏，例如 go get
```
是否应该写成 `go get`

freedomkk-qfeng replied:
可以有

inntran commented on runtime/golang.md:
```diff
@@ -0,0 +1,22 @@
+Golang 的生命周期策略为，支持最新的2个大版本。例如，Go 1.5在Go 1.7发布之前一直受支持，Go 1.6在Go 1.8发布之前一直受支持。
+
+### 兼容性
+Golang 1.x 版本保持向下兼容，因此任何低版本的 Go 语言程序都可以直接通过新版本直接编译来提升性能和修复 bug。
```
如何查看某个编译好的二进制使用哪个版本 Go 编译的？

freedomkk-qfeng replied:
golang 1.13 以后 go version 命令支持查看二进制版本。这个可以加一下

inntran commented on runtime/golang.md:
```diff
@@ -0,0 +1,22 @@
+
+|版本|发布日期|支持状态|
+|--|--|--|
+|Go1.14|2020/02/25|支持|
```
我看官网空格策略也很奇怪，我觉得 Go 和版本号之间加上空格读起来会容易一点。

freedomkk-qfeng replied:
实际上 go 官方的做法就是 go1.14 这样子，无论是官网还是代码和工具里。我用大写纯粹是个人喜好。现在想想可能还是应该改回小写

freedomkk-qfeng committed:
Update golang.md

1. 中英文保持空格
2. 增加查看 go 二进制的命令示例
3. 引入命令时，使用代码引用格式
4. 版本 release 部分，参照官网使用 go1.x 格式

freedomkk-qfeng committed:
Update golang.md

inntran commented:
LGTM
